### PR TITLE
[DO_NOT_MERGE] Add fuzz test for BinaryBuildRequestOptions

### PR DESCRIPTION
**Pull request by adambkaplan**
Does my conversion function do what it is supposed to do?

### CODE REVIEW

File: pkg/api/serialization_test.go [hunk 1]
```diff
@@ -259,6 +259,10 @@ func originFuzzer(t *testing.T, seed int64) *fuzz.Fuzzer {
 			c.FuzzNoCustom(j)
 			j.Type = image.SourceTagReferencePolicy
 		},
+		func(j *build.BinaryBuildRequestOptions, c fuzz.Continue) {
+			c.FuzzNoCustom(j)
+			j.CommitterEmail = ""
```
Comment (sttts) [inline pkg/api/serialization_test.go:264]: why this?

Comment (adambkaplan) [reply pkg/api/serialization_test.go]: @sttts I didn't know what I was doing at the time. I reverted the commit.

File: pkg/api/serialization_test.go [hunk 2]
```diff
@@ -561,6 +563,27 @@ func TestRoundTripDockerImage(t *testing.T) {
...
+		t.Errorf("failed to convert buildv1.BinaryBuildRequestOptions to url: %v", err)
+	}
+	reqOut := &buildv1.BinaryBuildRequestOptions{}
+	err = legacyscheme.Scheme.Convert(urlOut, reqOut, nil)
+	if err != nil {
+		t.Errorf("failed to convert URL to buildv1.BinaryBuildRequestOptions: %v", err)
+	}
+	if !reflect.DeepEqual(reqOut, reqIn) {
+		t.Errorf("Expected %s, got %s", reqIn, reqOut)
+	}
```
Comment (sttts) [inline pkg/api/serialization_test.go:583]: put a loop around and run it a hundred or thousand times, rule of thumb a second.

File: pkg/api/serialization_test.go [hunk 3]
```diff
@@ -561,6 +563,27 @@ func TestRoundTripDockerImage(t *testing.T) {
 	}
 }
 
+func TestBinaryBuildRequestOptions(t *testing.T) {
```
Comment (deads2k) [inline pkg/api/serialization_test.go:566]: I confirmed locally that this calls the conversion function.  Let's get this tidied up and passing and we'll merge

File: pkg/api/serialization_test.go (hunk 2 above)
Comment (adambkaplan) [reply pkg/api/serialization_test.go:603]: @sttts @deads2k cleaned up & runs 1000 times.

**Comment by adambkaplan** [comment]
/hold

This PR is for demonstration purposes only, and should not merge.

File: pkg/api/serialization_test.go [hunk 4]
```diff
@@ -561,6 +562,47 @@ func TestRoundTripDockerImage(t *testing.T) {
...
 }
 
+func TestBinaryBuildRequestOptionsFuzz(t *testing.T) {
+	seed := rand.Int63()
+	fuzzer := originFuzzer(t, seed)
+	for i := 0; i < 1000; i++ {
+		r := &buildv1.BinaryBuildRequestOptions{}
+		fuzzer.Fuzz(r)
+		urlOut := &url.Values{}
+		err := legacyscheme.Scheme.Convert(r, urlOut, nil)
```
Comment (sttts) [inline pkg/api/serialization_test.go:572]: nit: save lines with `if err := ...; err != nil`.

File: pkg/api/serialization_test.go [hunk 5]
```diff
@@ -561,6 +562,47 @@ func TestRoundTripDockerImage(t *testing.T) {
...
+		if err != nil {
+			t.Errorf("failed to convert buildv1.BinaryBuildRequestOptions to url.Values: %v", err)
+		}
+		decoded := &buildv1.BinaryBuildRequestOptions{}
+		err = legacyscheme.Scheme.Convert(urlOut, decoded, nil)
+		if err != nil {
+			t.Errorf("failed to convert url.Values to buildv1.BinaryBuildRequestOptions: %v", err)
+		}
+		// Converting to url.Values causes ObjectMeta data to be lost.
```
Comment (sttts) [inline pkg/api/serialization_test.go:581]: setting ObjectMeta to the zero value after fuzzing is fine IMO.

**Comment by adambkaplan** [comment]
/close

This proved correctness for #116, which merged.

**Pull request closed without merging**